c10/util/Exception.cpp

```diff
@@ -38,29 +38,50 @@ Error::Error(
               "] ",
               condition,
               ". ",
-              msg,
-              "\n"),
+              msg),
           backtrace,
           caller) {}
 
-std::string Error::msg() const {
-  return std::accumulate(
-             msg_stack_.begin(), msg_stack_.end(), std::string("")) +
-      backtrace_;
+std::string Error::compute_msg(bool include_backtrace) const {
+  if (msg_stack_.size() == 0) {
+    if (include_backtrace) return backtrace_;
+    return "";
+  }
+
+  std::ostringstream oss;
+
+  if (msg_stack_.size() == 2) {
+    // Fold error and context in one line
+    oss << msg_stack_[0] << " (" << msg_stack_[1] << ")";
+  } else {
+    oss << msg_stack_[0];
+    for (size_t i = 1; i < msg_stack_.size(); i++) {
```

Comment:
nit: auto i = decltype(msg_stack_.size()){1}

Reply:
Come on, is this really what C++ programmers are supposed to write? XD

Reply:
There's some level of hating build warnings where it becomes right?